### ISSUE TITLE
Not showing all git submodules

### ISSUE THREAD

**Issue by elnicoCZ** [microsoft/vscode#100928]
Issue Type: <b>Bug</b>

I have a git repo with 13 submodules. All of these are displayed with `git submodule status`. Anyway, only 10 of them are displayed within the _SOURCE CONTROL PROVIDERS_ list, together with the master repository on the first line (11 in total).

When I make some changes in one of the invisible repos, the master repository shows there is a change in the submodule - the submodule is listed in the _CHANGES_ list. Anyway, when I click this line, no diff is opened. Normally, I would get a diff like follows:

```diff
-Subproject commit e6d948676f925f591fe612b95cf41e7537b28abc
+Subproject commit e6d948676f925f591fe612b95cf41e7537b28abc-dirty
```

Note: I am aware of the _Scm › Providers: Visible_ setting. The issue is not that the submodules would be scrolled out. They are simply not there.

VS Code version: Code 1.46.1 (cd9ea6488829f560dc949a8b2fb789f3cdc05f5d, 2020-06-17T21:13:08.304Z)
OS version: Linux x64 4.19.0-9-amd64
Remote OS version: Linux x64 4.15.0-106-generic

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (4 x 2592)|
|GPU Status|2d_canvas: unavailable_software<br>flash_3d: disabled_software<br>flash_stage3d: disabled_software<br>flash_stage3d_baseline: disabled_software<br>gpu_compositing: disabled_software<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: disabled_off<br>rasterization: disabled_software<br>skia_renderer: disabled_off_ok<br>video_decode: disabled_software<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off_ok<br>webgl: unavailable_software<br>webgl2: unavailable_software|
|Load (avg)|0, 0, 0|
|Memory (System)|3.85GB (0.28GB free)|
|Process Argv|--no-sandbox|
|Screen Reader|no|
|VM|100%|
|DESKTOP_SESSION|/usr/share/xsessions/lxqt|
|XDG_CURRENT_DESKTOP|LXQt|
|XDG_SESSION_DESKTOP||
|XDG_SESSION_TYPE|x11|

|Item|Value|
|---|---|
|Remote|SSH: pilatus|
|OS|Linux x64 4.15.0-106-generic|
|CPUs|Intel(R) Core(TM) i7-6820HQ CPU @ 2.70GHz (8 x 2999)|
|Memory (System)|15.55GB (4.24GB free)|
|VM|0%|
</details><details><summary>Extensions (18)</summary>

Extension|Author (truncated)|Version
---|---|---
doxdocgen|csc|0.6.0
vscode-markdownlint|Dav|0.36.0
gitlens|eam|10.2.2
todo-tree|Gru|0.0.177
kconfig|luv|1.0.3
camelcasenavigation|map|1.1.2
git-graph|mhu|1.24.0
vscode-docker|ms-|1.3.1
remote-containers|ms-|0.122.1
remote-ssh|ms-|0.51.0
remote-ssh-edit|ms-|0.51.0
cmake-tools|ms-|1.4.1
cpptools|ms-|0.28.3
hexeditor|ms-|1.0.1
devicetree|plo|0.1.1
diff|raf|0.0.1
shellcheck|tim|0.9.0
cmake|twx|0.0.17

</details>

**Comment by joaomoreno** [comment]
Please increase the limit using the `git.detectSubmodulesLimit` setting.

**Comment by elnicoCZ** [comment]
Hi Joao, thank you, that was indeed the problem.

Anyway, I suggest there is a space for improvement. I understand it may be reasonable to have some limit, but how about to show a warning in the _SOURCE CONTROL PROVIDERS_ list that not all submodules are loaded, with a link to the appropriate setting? I admit I did not spend _much_ time with looking for the setting, on the other hand I did spend _some_ time - I searched for _provider_. But how can I, as a user, know that this is a problem of the git extension so I should be looking for git settings?
